File: test/unit/data/set_sdk_auth_token_payload_test.dart
import 'package:exponea/src/data/normalization/set_sdk_auth_token_payload.dart';
import 'package:flutter_test/flutter_test.dart';

import 'base.dart';

void main() {
  group('encodeSetSdkAuthTokenPayload', () {
    final data = readMapData('set_sdk_auth_token_payload');

    test('check data', () {
      expect(data.length, 1);
    });

    test('encodes token', () {
      expect(
        encodeSetSdkAuthTokenPayload('mock-jwt-token'),
        data[0],
      );
    });
  });
}

File: example/android/app/src/test/kotlin/com/exponea/example/SetSdkAuthTokenPayloadTest.kt
package com.exponea.example

import org.junit.Assert.assertEquals
import org.junit.BeforeClass
import org.junit.Test

class SetSdkAuthTokenPayloadTest {
    companion object {
        lateinit var data: List<Map<String, Any?>>

        @BeforeClass @JvmStatic fun setup() {
            data = BaseTest.readMapData("set_sdk_auth_token_payload")
        }
    }

    @Test
    fun `validate data`() {
        assertEquals(data.size, 1)
    }

    @Test
    fun `parse token payload`() {
        assertEquals(data[0]["token"], "mock-jwt-token")
    }
}
